rfcs/0000-expressive-diagnostics.md

```diff
@@ -0,0 +1,356 @@
+- Feature Name: `expressive_diagnostics`
+- Start Date: 2023-08-07
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
+- Sway Issue: [FueLabs/sway#0000](https://github.com/FuelLabs/sway/issues/001)
```

Comment:
Is there a sway issue for this RFC?

Reply:
As far as I understood the RFC process, the name gets renamed and the links get set after the PR is accepted. The PR accepts commits after it is approved so I'll adjust the links and the file name once we agree on the content. So, dive in, looking forward to your comments :-)

Reply:
I'll also create the Sway issue before adjusting the RFC.

Reply:
Ahh okay, my bad then!